src/lib.rs

```diff
@@ -525,10 +510,45 @@ where
         }
     }
 
-    while let Some(msg) = b.next() {
+    if let Some(log) = cfg.logger.as_ref() {
+        info!(log, "compiled all datapath programs, ccp ready");
+    }
+
+    while let Some((msg, recv_addr)) = b.next() {
         match msg {
+            Msg::Rdy(r) => {
+                if dp_to_flowmap.remove(&recv_addr).is_some() {
+                    if let Some(log) = cfg.logger.as_ref() { 
+                        info!(log, "new ready from old datapath, clearing old flows and installing programs...");
```

Comment:
what is "old datapath"?

Reply:
ie. one we've already seen before (maybe the datapath died and restarted or something)

Reply:
if we get a new ready message from a datapath we've seen before, we assume it wants fresh state, so we'll send the datapath programs over again and clear any flow state we had for this datapath. this allows the datapath to restart without having to restart ccp. the purpose of the message is just a warning that it is clearing out old state from a previous run